features/0593-json-ld-cred-attach/README.md

```diff
@@ -0,0 +1,206 @@
+
+## Tutorial
+
+Complete examples of messages are provided in the [reference section](#reference).
+
+## Reference
+
+### `ld-proof-vc-proposal` attachment format
+
+Format identifier: `aries/ld-proof-vc-proposal@v1.0`
+
+The credential proposal allows the holder to initiate a credential exchange without fully specifying the contents of the credential. If all properties of the credential are already known, the protocol can also be initiated with a credential request. While the credential detail requires all fields to be present, the proposal allows for fields to be omitted.
+
+The below `propose-credential` message shows an example of a credential proposal with the `@context`, `type`, `credentialSubject` and `credentialSubject` present. The issuer can then send a credential detail in a credential offer message adding extra properties required for issuance such as `issuer`, `credentialStatus.type`, `proofType`, etc..
+
+```json
+{
+  "@id": "257363ab-a3f9-4bba-ad3f-9f69501135f5",
+  "@type": "https://didcomm.org/issue-credential/%VER/propose-credential",
+  "comment": "<some comment>",
+  "formats": [
+    {
+      "attach_id": "35054fa5-112e-4ef7-93ab-7778f59eb4e3",
+      "format": "aries/vc-ld-proof@v1.0"
+    }
+  ],
+  "filter~attach": [
+    {
+      "@id": "35054fa5-112e-4ef7-93ab-7778f59eb4e3",
+      "mime-type": "application/ld+json",
+      "data": {
+        "json": {
+          "@context": [
+            "https://www.w3.org/2018/credentials/v1",
+            "https://www.w3.org/2018/credentials/examples/v1"
+          ],
+          "type": ["VerifiableCredential", "UniversityDegreeCredential"],
+          "credentialSubject": {
+            "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+            "degree": {
+              "type": "BachelorDegree",
+              "name": "Bachelor of Science and Arts"
+            }
+          },
+          "credentialSchema": {
+            "id": "https://example.org/examples/degree.json",
+            "type": "JsonSchemaValidator2018"
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+- `credentialStatus.id` -- Credential Status may be omitted completely, but if present should contain the `type` property to indicate which mechanism to use. As the id can be closely tied to the issuance process the property is not required for the credential proposal.
+- `proofType` -- This is not a property defined by the VC data model, but indicates the proof types the holder proposes for the credential. Entries match the `type` property of the [Proofs](https://www.w3.org/TR/vc-data-model/#proofs-signatures) object. Multiple entries indicate the holder proposes multiple proofs for the credential.
```

Comment:
Registry for LD proof types: https://w3c-ccg.github.io/ld-cryptosuite-registry/

Reply:
This also seems like a good candidate for an `options` object.

Reply:
Moved `proofType` to options and added link to registry

Reply:
Note: although its presence is required by the VC data model, this property definitely makes sense to NOT be included in the credential "template" in our use case for obvious reasons